crates/sui-json-rpc/src/error.rs

```diff
@@ -399,10 +400,10 @@ mod tests {
             let error_object: ErrorObjectOwned = rpc_error.into();
             let expected_code = expect!["-32002"];
             expected_code.assert_eq(&error_object.code().to_string());
-            let expected_message = expect!["Failed to sign transaction by a quorum of validators because of locked objects. Retried a conflicting transaction Some(TransactionDigest(11111111111111111111111111111111)), success: Some(true)"];
+            let expected_message = expect!["Failed to sign transaction by a quorum of validators because of locked objects. Retried a conflicting transaction 8qbHbw2BbbTHBW1sbeqakYXVKRQM8Ne7pLK7m6CVfeR, success: true. Conflicting Transactions: [4vJ9JU1bJJE96FWSJKvHsmmFADCg4gpZQff4P3bkLKi]"];
```

Comment:
- Do you know whether usually the `conflicting_txes` map includes the transaction that was retried or not? It would be good to get the test to look like the "real thing" as much as possible.
- One useful piece of information that's missing here is the weight of each conflicting transaction (how much of the stake by weight voted for each transaction -- `stake_unit` in the code). Can we order conflicting transactions in decreasing order of stake weight and include their stake weight as a percentage in the message? Stake units are given in basis points. Having this information will help identify whether the object is equivocated, or whether it is used for something else.
- We can incorporate the `success: true` into the message a bit more -- do you know whether it's possible to extract more information about the conflicting transaction's execution status (if it also errored out, can we provide its error?)
- Let's add some test cases for other errors in this same family (object double used):
  - Tried a conflicting transaction and it failed.
  - Multiple conflicting transactions.
  - Could not retry a conflicting transaction (does this imply that the object is equivocated? If so, can we make that more obvious in the error message?).

Reply:
> Do you know whether usually the conflicting_txes map includes the transaction that was retried or not? It would be good to get the test to look like the "real thing" as much as possible.

I just tried it and yeah this does appear to be the case.. A little confused by this because the way I set this up is by making two transactions that should hash differently (one transfer 1 coin the other moves 2 coins) so how are these digests the same my understanding was that the conflicting transaction is different from the retried.
```
JsonRpcError: Failed to sign transaction by a quorum of validators because of locked objects. Retried a conflicting transaction CoQQ96bjA5UVKhG7sFqjAPmCgcTf7kRY9W9jWHnsasNk, success: true. Conflicting Transactions: [CoQQ96bjA5UVKhG7sFqjAPmCgcTf7kRY9W9jWHnsasNk]
```

> Order conflicting transactions in decreasing order of stake

I'll implement this

> do you know whether it's possible to extract more information about the conflicting transaction's execution status

That might require a good amount of pluming since each usage of this error might not have that information available when the error is created. I'll take a look